### DISCUSSION TITLE
Const Correcness

### DISCUSSION THREAD

**Discussion by TobiasWallner** [jupyter-xeus/cpp-terminal#264] [General]
Hi, I was recently working with the event types and wanted to use them as a const reference for a function argument, but it cannot really be used in a `const` environment, because all its methods lack `const` qualifiers. (besides the fact, that: yes of course, you can just copy or move it instead). I don't know about you but I like to work in `const` wherever I can.

would you like it if I added them in a pull-request?

files and lines that I would change:
* https://github.com/jupyter-xeus/cpp-terminal/blob/0674a799ba7ab99fab0bae48e35415aa23d3f1fb/cpp-terminal/event.hpp#L28-L34
* https://github.com/jupyter-xeus/cpp-terminal/blob/0674a799ba7ab99fab0bae48e35415aa23d3f1fb/cpp-terminal/key.hpp#L198-L226
* https://github.com/jupyter-xeus/cpp-terminal/blob/0674a799ba7ab99fab0bae48e35415aa23d3f1fb/cpp-terminal/screen.hpp#L14-L17

**Comment by flagarde** [comment]
I fully agree with you. I like it too !

**Comment by flagarde** [comment]
Merged, Thx !
